packages/api/src/hooks/p2p/useAdvertiserAdverts.ts

```diff
@@ -0,0 +1,68 @@
+import { useMemo } from 'react';
+import useInfiniteQuery from '../../useInfiniteQuery';
+import useAuthorize from '../useAuthorize';
+
+/** This custom hook returns a list of adverts under the current active client. */
+const useAdvertiserAdverts = (
+    payload?: NonNullable<Parameters<typeof useInfiniteQuery<'p2p_advertiser_adverts'>>[1]>['payload'],
+    config?: NonNullable<Parameters<typeof useInfiniteQuery<'p2p_advertiser_adverts'>>[1]>['options']
+) => {
+    const { isSuccess } = useAuthorize();
+    const { data, fetchNextPage, ...rest } = useInfiniteQuery('p2p_advertiser_adverts', {
+        payload: { ...payload, offset: payload?.offset, limit: payload?.limit },
+        options: {
+            ...config,
+            getNextPageParam: (lastPage, pages) => {
+                if (!lastPage?.p2p_advertiser_adverts?.list) return;
+
+                return pages.length;
+            },
+            enabled: isSuccess && (config?.enabled === undefined || config.enabled),
+        },
+    });
+
+    const flatten_data = useMemo(() => {
+        if (!data?.pages?.length) return;
+
+        return data?.pages?.flatMap(page => page?.p2p_advertiser_adverts?.list);
+    }, [data?.pages]);
+
+    const modified_data = useMemo(() => {
+        if (!flatten_data?.length) return undefined;
+
+        return flatten_data.map(advert => ({
+            ...advert,
+            /** Determine if the rate is floating or fixed */
+            is_floating: advert?.rate_type === 'float',
```

Comment:
Since you are checking the undefined data you don't need to pass optional chain here and for the rest of the keys/values in this obj
```suggestion
            is_floating: advert.rate_type === 'float',
```

Reply:
The value inside the flatten_map can still be undefined because the return value from flatten map can be p2p_order_list[] or undefined[] @niloofar-deriv 🙏 

![image](https://github.com/binary-com/deriv-app/assets/125247833/39e4800b-9e54-44da-98da-2dac9d01647b)